File: Backend/SmartMicrogrid.API/DTOs/Transactions/GenerateQrResponse.cs
namespace SmartMicrogrid.API.DTOs.Transactions
{
    public class GenerateQrResponse
    {
        public string TransactionId { get; set; } = string.Empty;

        public string ReservationId { get; set; } = string.Empty;

        public string TransactionCode { get; set; } = string.Empty;

        public string QrCodeData { get; set; } = string.Empty;

        public string Status { get; set; } = string.Empty;
    }
}

File: Android/SmartMicrogrid.Android/app/src/main/java/com/smartmicrogrid/models/GenerateQrResponse.kt
package com.smartmicrogrid.models

import com.google.gson.annotations.SerializedName

data class GenerateQrResponse(
    @SerializedName("transactionId")
    val transactionId: String = "",

    @SerializedName("reservationId")
    val reservationId: String = "",

    @SerializedName("transactionCode")
    val transactionCode: String = "",

    @SerializedName("qrCodeData")
    val qrCodeData: String = "",

    @SerializedName("status")
    val status: String = ""
)